ixwebsocket/IXWebSocket.cpp

```diff
@@ -293,8 +293,8 @@ namespace ix
                 break;
             }
 
-            // We cannot enter poll which might block forever if we are stopping
-            if (_stop) break;
+            // We can avoid to poll if we want to stop and are not closing
+            if (_stop && !isClosing()) break;
```

Comment:
The bug I fixed the other day (Wednesday) explicitely want that break clause WHEN closing, otherwise we were hanging on macOS on the CI.

Reply:
It seems not, the macOS Ci is green

Reply:
And on my Mac too :)

Reply:
Anyway if we don’t do this, no close callback at all are called, because we are closing and expected to deal with this situation in the poll or in the dispatch

Reply:
If no close callbacks are ever called indeed that's a serious bug. I will test that again. 
The CI has been extremely unstable, the macOS CI would randomly fail.

Reply:
I pushed that commit to run the windows CI unittest on one thread 
aa3f201cedfb4bc0697b1552eacd35afb88313b5

That will let us see the error message in the close unittest.

Reply:
That’s why the test fails with 0 as client close code, which means not close code set in the test so no close callback called; confirmed by no “client disconnect” test log.
That’s the problem of disabling tests and adding break conditions, before going deeper in the investigation I expected some kind of trouble like that. We don’t have to forget that close() method just send the close frame and set to closing state, but we have to wait for the close frame of the server or deals with a connection closed, so need to poll and dispatch, and if finally nothing is received the closeMaximim delay will close the connection.

Reply:
Will it rerun again?